op-e2e/actions/op_program_test.go

```diff
@@ -0,0 +1,270 @@
+package actions
+
+import (
+	"context"
+	"math/rand"
+	"os"
+	"testing"
+
+	batcherFlags "github.com/ethereum-optimism/optimism/op-batcher/flags"
+	"github.com/ethereum-optimism/optimism/op-e2e/e2eutils"
+	"github.com/ethereum-optimism/optimism/op-program/host"
+	"github.com/ethereum-optimism/optimism/op-program/host/config"
+	"github.com/ethereum-optimism/optimism/op-service/sources"
+	"github.com/ethereum-optimism/optimism/op-service/testlog"
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/ethereum/go-ethereum/common/hexutil"
+	"github.com/ethereum/go-ethereum/core/types"
+	"github.com/ethereum/go-ethereum/log"
+	"github.com/stretchr/testify/require"
+)
+
+var dumpFixtures = false
+
+func init() {
+	if os.Getenv("OP_E2E_DUMP_FIXTURES") == "1" {
+		dumpFixtures = true
+	}
+}
```

Comment:
Can ignore this for now - will follow up with another PR for automatic [fp-test](https://github.com/ethereum-optimism/fp-tests) fixture generation from these tests.